acceptance-tests/domain_fronting_test.go

```diff
@@ -171,6 +153,9 @@ var _ = Describe("Domain fronting", func() {
 			expect421(nonMTLSClient.Do(req))
 			expect421(mtlsClient.Do(req))
 
+			By("Sending a request to HAProxy with a mismatched SNI and Host header it returns a 400")
+			expect400BadRequestNoHostHeader(haproxyInfo.PublicIP, tlsConfig)
+
```

Comment:
We should have one additional test:
- domain fronting ENABLED (i.e. disableDomainFronting = false )
- Sending a request to HAproxy with no Host header it returns a 400

Just to make sure that the old logic is still intact. (beware: the old logic does not have a body so the `expect400BadRequestNoHostHeader` function will probably fail.)

The old response:
```
HTTP/1.1 400 Bad Request: missing required Host header
content-type: text/plain; charset=utf-8
connection: close

```
So the message is part of the status code instead of the body.

Reply:
I'll add it but note that this will add a few minutes to the acceptance tests